builtin/mainmenu/tab_singleplayer.lua

```diff
@@ -38,6 +38,14 @@ local function singleplayer_refresh_gamebar()
 					core.set_topleft_text(gamemgr.games[j].name)
 					core.setting_set("menu_last_game",gamemgr.games[j].id)
 					menudata.worldlist:set_filtercriteria(gamemgr.games[j].id)
+					local index = filterlist.get_current_index(menudata.worldlist,
+						tonumber(core.setting_get("mainmenu_last_selected_world")))
+					local selected = core.get_textlist_index("sp_worlds")
+					if not index or index < 1 then
+						index = math.min(core.get_textlist_index("sp_worlds"),
+							#menudata.worldlist:get_list())
+					end
+					menu_worldmt_legacy(index)
```

Comment:
ah dammit yea you are right
if its "moved up", it would need to get into the else case.